engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game.console.commands;
+
+import org.destinationsol.files.HullConfigManager;
+import org.destinationsol.game.Console;
+import org.destinationsol.game.Hero;
+import org.destinationsol.game.SolGame;
+import org.destinationsol.game.console.ConsoleInputHandler;
+import org.destinationsol.game.screens.ChangeShipScreen;
+import org.destinationsol.game.ship.ShipRepairer;
+import org.destinationsol.game.ship.SolShip;
+import org.destinationsol.game.ship.hulls.HullConfig;
+
+import java.util.Optional;
+
+/**
+ * A command used to change current ship on the fly.
+ * Takes one argument - id of the ship you want to load
+ */
+public class ChangeShipCommandHandler implements ConsoleInputHandler {
+
+    private Hero hero;
+    private SolGame game;
+
+    public ChangeShipCommandHandler(Hero hero, SolGame game) {
+        this.hero = hero;
+        this.game = game;
+    }
+
+    @Override
+    public void handle(String input, Console console) {
+        String[] args = input.split(" ", 2);
+
+        if(args.length != 2) {
+            printHelp(console);
+            return;
+        }
+
+        Optional<SolShip> newShip = cloneAndModifyShip(hero.getShip(), args[1]);
+        if(!newShip.isPresent()) {
+            printHelp(console);
+            return;
+        }
+
+        game.getObjectManager().removeObjDelayed(hero.getShip());
+        game.getObjectManager().addObjDelayed(newShip.get());
+        hero.setSolShip(newShip.get(), game);
+    }
+
+    private void printHelp(Console console) {
+        console.warn("Invalid or Unknown ship ID.");
+        console.warn("Usage: \"changeship module:shipName\"");
+    }
+
+    private Optional<SolShip> cloneAndModifyShip(SolShip originalShip, String newShipID) {
+        HullConfig newHullConfig;
+        try {
+            HullConfigManager hcm = game.getHullConfigManager();
+            newHullConfig = hcm.getConfig(newShipID);
+        }
+        catch (RuntimeException e) {
+            System.out.println(e.getMessage());
```

Comment:
Please use the logger instead of System.out

Reply:
You could also go around this whole try/catch by using `Assets.getDefaultAssetHelper().listJSONByRegex()` to ensure the existence of the ship, and `Validater.getValidatedJSON()` wit hthe proper schema should ensure you get really a ship. If it excepts even afterward, it's almost definitely a broken ship then, and you should crash in that case.

Reply:
Please note - I have no idea how are the functions and classes mentioned in the above comment actually named.  The names I wrote just look righty-ish. You should be able to find them though if you can look around a bit.